x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -1414,7 +1415,8 @@ private static class ResolveInference extends ParameterizedRule<LogicalPlan, Log
 
         @Override
         public LogicalPlan apply(LogicalPlan plan, AnalyzerContext context) {
-            return plan.transformDown(InferencePlan.class, p -> resolveInferencePlan(p, context));
+            return plan.transformDown(InferencePlan.class, p -> resolveInferencePlan(p, context))
+                .transformExpressionsOnly(InferenceFunction.class, f -> resolveInferenceFunction(f, context));
```

Comment:
Non-blocking question for my ES|QL education: Can you help me understand why this change?

Reply:
```
plan.transformDown(InferencePlan.class, p -> resolveInferencePlan(p, context))
```
will transform all the children in the plan that have the class `InferencePlan` with the result of the `resolveInferencePlan(p, context)` call.  Inference plans are typically command using inference: `RERANK` and `COMPLETION`

```
.transformExpressionsOnly(InferenceFunction.class, f -> resolveInferenceFunction(f, context));

```
will transform do the same but for `InferenceFunction` instead of plan. Because text embedding is our first inference function this was not yet done, so I added it.